affinity/cpp-20/d0796r1.md

```diff
@@ -225,35 +261,83 @@ The `execution_context` class provides an abstraction for managing a number of l
 
 ### `execution_context` member aliases
 
-	using executor_type = __unspecfied__;
+    using executor_type = __unspecfied__;
 
 *Requires:*
 
 ### `execution_context` constructors
 
-	template <typename ExecutionResource>
-	execution_context(ExecutionResource &&execResource);
+    execution_context(const execution_resource &);
 
 ### `execution_context` destructor
-
+    
     ~execution_context();
 
 ### `execution_context` operators
 
-	const execution_resource &resource() const noexcept;
+    const execution_resource &resource() const noexcept;
 
 *Returns:*
 
-	executor_type executor() noexcept;
+    executor_type executor() noexcept;
 
 *Returns:*
 
-## Free functions
+    pmr::memory_resource *memory_resource() noexcept;
+
+*Returns:*
+
+## Class template `affinity_query`
 
-	const this_system::execution_resource &resource();
+The `affinity_query` class template provides an abstraction for a relative affinity value between two `execution_resource`s, derived from a particular `affinity_operation` and `affinity_metric`.
+
+### `affinity_query` member aliases
+
+    using native_affinity_type = __unspecfied__;
+
+*Requires:*
+
+    using error_type = __unspecfied__;
+
+*Requires:*
+
+### `affinity_query` constructors
+
+    affinity_query(const execution_resource &, const execution_resource &);
+
+### `affinity_query` destructor
+
+    ~affinity_query();
+
+### `affinity_query` operators
+
+    native_affinity_type native_affinity() const noexcept;
+
+*Returns:* Unspecified native affinity value.
+
+### `affinity_query` comparisons
+
+    friend expected<size_t> operator==(const affinity_query&, const affinity_query&);
+    friend expected<size_t> operator!=const affinity_query&, const affinity_query&);
+    friend expected<size_t> operator<(const affinity_query&, const affinity_query&);
+    friend expected<size_t> operator>(const affinity_query&, const affinity_query&);
+    friend expected<size_t> operator<=(const affinity_query&, const affinity_query&);
+    friend expected<size_t> operator>=(const affinity_query&, const affinity_query&);
 
 *Returns:*
 
+> [*Note:* The comparison operators rely on the availability of the `expected` class template (see [P0323r4: std::expected][p0323r4]), if this does not become available then an alternative error/value construct will be adopted instead. *--end note*]
+
+## Free functions
+
+	std::vector<execution_resource> resources() noexcept;
+
+*Returns:* A std::vector containing all system level resources.
+
+*Requires:* If `resources().size() > 0`, `resources()[0]` be the `execution_resource` corroponding to the current thread of execution. The value returned by `resources()` be the same at any point after the invocation of `main`.
```

Comment:
I think we should require resources().size() to be at least 1.

Reply:
This is what we require now, or do you mean we should say `>= 1` rather than `> 0`?